fix(with zones): params causes a failure to compile

Kaikaikaifang opened the pull request:
Type '{ params: { id: any; }; }' does not satisfy the constraint 'PageProps'.

### What?

Fix example: with zones bug

![image](https://github.com/user-attachments/assets/f003d9fb-07c2-4c31-8d1d-1dbc6218ae88)

### Why?

Uncompilable due to nextjs update

### How?

Adapt to nextjs15:

await params

Kaikaikaifang committed:
fix(with zones): params causes a failure to compile

Type '{ params: { id: any; }; }' does not satisfy the constraint 'PageProps'.

dominictracey commented:
I was just getting ready to do this. Thanks @Kaikaikaifang

dominictracey commented:
Though I got slightly different code from `npx @next/codemod@latest --force next-async-request-api .`, which might be more appropriate.

```
type Params = {
  params: Promise<{
    slug: string;
  }>;
};

export async function generateMetadata(props: Params): Promise<Metadata> {
  const params = await props.params;
```

Kaikaikaifang committed:
style: maintain the style with docs

https://nextjs.org/docs/app/building-your-application/upgrading/version-15#asynchronous-page

Kaikaikaifang commented:
> Though I got slightly different code from `npx @next/codemod@latest --force next-async-request-api .`, which might be more appropriate.
> 
> ```
> type Params = {
>   params: Promise<{
>     slug: string;
>   }>;
> };
> 
> export async function generateMetadata(props: Params): Promise<Metadata> {
>   const params = await props.params;
> ```

You are right.
I changed the code to keep it in sync with the [documentation](https://nextjs.org/docs/app/building-your-application/upgrading/version-15#asynchronous-page) style

Kaikaikaifang committed:
Merge branch 'canary' into canary

Kaikaikaifang committed:
Merge branch 'canary' into canary

Kaikaikaifang commented:
@dominictracey 

I noticed that the example with zones have not been updated and this PR is still going on, please reply!

Kaikaikaifang commented:
Nobody care.
